ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/ClientTest.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client;
+
+import static io.confluent.ksql.test.util.AssertEventually.assertThatEventually;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+
+import io.confluent.ksql.api.BaseApiTest;
+import io.confluent.ksql.api.client.impl.ClientImpl;
+import io.confluent.ksql.api.client.impl.ClientOptionsImpl;
+import io.confluent.ksql.api.server.PushQueryId;
+import io.confluent.ksql.parser.exception.ParseFailedException;
+import io.vertx.core.Context;
+import io.vertx.ext.web.client.WebClient;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import org.junit.Test;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+public class ClientTest extends BaseApiTest {
+
+  @SuppressWarnings("unchecked")
+  protected static final List<String> DEFAULT_COLUMN_NAMES = BaseApiTest.DEFAULT_COLUMN_NAMES.getList();
+  @SuppressWarnings("unchecked")
+  protected static final List<String> DEFAULT_COLUMN_TYPES = BaseApiTest.DEFAULT_COLUMN_TYPES.getList();
+  protected static final Map<String, Object> DEFAULT_PUSH_QUERY_REQUEST_PROPERTIES =
+      BaseApiTest.DEFAULT_PUSH_QUERY_REQUEST_PROPERTIES.getMap();
+
+  protected Context context;
+  protected Client client;
+
+  @Override
+  public void setUp() {
+    super.setUp();
+
+    context = vertx.getOrCreateContext();
+  }
+
+  @Override
+  protected WebClient createClient() {
+    // Use Java client for these tests, rather than a vanilla WebClient
+    this.client = createJavaClient();
```

Comment:
Perhaps use a different method for this, as it's doing something different to creating a WebClient

Reply:
Done.